src/Arcus.Security.Core/Providers/MutatedSecretNameCachedSecretProvider.cs

```diff
@@ -0,0 +1,139 @@
+        /// </summary>
+        public ICacheConfiguration Configuration => _implementation.Configuration;
+
+        /// <summary>
+        /// Retrieves the secret value, based on the given name
+        /// </summary>
+        /// <param name="secretName">The name of the secret key</param>
+        /// <param name="ignoreCache">Indicates if the cache should be used or skipped</param>
+        /// <returns>Returns a <see cref="Task{TResult}"/> that contains the secret key</returns>
+        /// <exception cref="ArgumentException">The name must not be empty</exception>
+        /// <exception cref="ArgumentNullException">The name must not be null</exception>
+        /// <exception cref="SecretNotFoundException">The secret was not found, using the given name</exception>
+        public async Task<string> GetRawSecretAsync(string secretName, bool ignoreCache)
+        {
+            Guard.NotNullOrWhitespace(secretName, nameof(secretName), "Requires a non-blank secret name when mutating secret names");
+
+            string mutatedSecretName = MutateSecretName(secretName);
+            Task<string> rawSecretAsync = _implementation.GetRawSecretAsync(mutatedSecretName, ignoreCache);
+
+            if (rawSecretAsync is null)
+            {
+                throw new SecretNotFoundException(mutatedSecretName);
+            }
+
+            try
+            {
+                return await rawSecretAsync;
+            }
+            catch (Exception exception)
+            {
+                Logger.LogError(
+                    exception, "Failure during retrieving secret '{MutatedSecretName}' that was mutated from '{OriginalSecretName}'", mutatedSecretName, secretName);
+
+                throw new SecretNotFoundException(mutatedSecretName, exception);
+            }
+        }
+
+        /// <summary>
+        /// Retrieves the secret value, based on the given name
+        /// </summary>
+        /// <param name="secretName">The name of the secret key</param>
+        /// <param name="ignoreCache">Indicates if the cache should be used or skipped</param>
+        /// <returns>Returns a <see cref="Task{TResult}"/> that contains the secret key</returns>
+        /// <exception cref="ArgumentException">The name must not be empty</exception>
+        /// <exception cref="ArgumentNullException">The name must not be null</exception>
+        /// <exception cref="SecretNotFoundException">The secret was not found, using the given name</exception>
+        public async Task<Secret> GetSecretAsync(string secretName, bool ignoreCache)
+        {
+            Guard.NotNullOrWhitespace(secretName, nameof(secretName), "Requires a non-blank secret name when mutating secret names");
+
+            string mutatedSecretName = MutateSecretName(secretName);
+            Task<Secret> secretAsync = _implementation.GetSecretAsync(mutatedSecretName, ignoreCache);
+
+            if (secretAsync is null)
+            {
```

Comment:
Same here, correct exception or not?

Reply:
Changed exception type and moved to central methd. Resolving.